src/mixins/hasModal.js

```diff
@@ -0,0 +1,18 @@
+export default {
+  data() {
+    return {
+      isModalOpen: false,
+    }
+  },
+  beforeRouteLeave(to, from, next) {
+    if (this.isModalOpen) {
+      document.querySelector('ion-modal-controller').dismiss()
```

Comment:
this really is ugly

Reply:
does it make sense to query from the component's top-level element instead of `document`?

Reply:
Since this mixin is not attached to the Modal.vue file, we can't really be sure where the controller is inside the document, but this will be abstracted into `ionic-vue` down the road though